types/cloudflare/index.d.ts

```diff
@@ -0,0 +1,287 @@
+            tags?: string[];
+            hosts: string[];
+        },
+    ): Promise<any>;
+    browse(): Promise<any>;
+}
+
+export interface ZoneSettings {
+    read(id: string, setting: string): Promise<any>;
+    edit(id: string, setting: string, value: string | Record<string, unknown>): Promise<any>;
+    editAll(id: string, settings: any): Promise<any>;
+    browse(id: string): Promise<any>;
+}
+
+export interface ZoneCustomHostNames {
+    browse(zone_id: string): Promise<any>;
+    read(zone_id: string, id: string): Promise<any>;
+    edit(
+        zone_id: string,
+        id: string,
+        config: {
+            hostname: string;
+            ssl: {
+                method: string;
+                type: string;
+                settings: {
+                    http2: string;
+                    http3: string;
+                    min_tls_version: string;
+                    tls_1_3: string;
+                    ciphers: string[];
+                };
+                bundle_method: string;
+                wildcard: boolean;
+                custom_certificate: string;
+                custom_key: string;
+            };
+        },
+    ): Promise<any>;
+    add(
+        zone_id: string,
+        config: {
+            hostname: string;
+            ssl: {
+                method: string;
+                type: string;
+                settings: {
+                    http2: string;
+                    http3: string;
+                    min_tls_version: string;
+                    tls_1_3: string;
+                    ciphers: string[];
+                };
+                bundle_method: string;
+                wildcard: boolean;
+                custom_certificate: string;
+                custom_key: string;
+            };
+        },
+    ): Promise<any>;
+    del(zone_id: string, id: string): Promise<any>;
+}
+
+export interface ZoneWorkers {
+    validate(zone_id: string, script: string): Promise<any>;
+}
+
+export interface ZoneWorkersScript {
+    read(zone_id: string, script?: string): Promise<any>;
+    del(): Promise<any>;
+}
+
+export interface ZoneWorkersRoutes {
+    browse(zone_id: string): Promise<any>;
+    edit(zone_id: string, id: string, config: { pattern: string; script: string }): Promise<any>;
+    read(zone_id: string, id: string): Promise<any>;
+    add(zone_id: string, config: { pattern: string; script: string }): Promise<any>;
+    del(zone_id: string, id: string): Promise<any>;
+}
+
+export interface User {
+    read(): Promise<any>;
+    edit(user: {
+        first_name: string;
+        last_name: string;
+        telephone: string;
+        country: string;
+        zipcode: string;
+    }): Promise<any>;
+}
+
+export interface Stream {
+    listVideos(accountId: string): Promise<any>;
+    videoDetails(accountId: string, id: string): Promise<any>;
+    deleteVideo(accountId: string, id: string): Promise<any>;
+}
+
+export class Cloudflare {
```

Comment:
This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like in the following:

```ts
class Cloudflare {
  // ...
}

namespace Cloudflare {
    // all other type declarations go here so they're still accessible
}

export = Foo
```

To import the entity, you'd want to use syntax like

```ts
import Cloudflare = require("cloudflare");
```